sentry-spring/src/main/java/io/sentry/spring/webflux/SentryWebFilter.java

```diff
@@ -96,7 +96,7 @@ isTracingEnabled && shouldTraceRequest(requestHub, request)
             () -> {
               serverWebExchange.getAttributes().put(SENTRY_SCOPES_KEY, requestHub);
               Sentry.setCurrentScopes(requestHub);
-              requestHub.pushScope();
+              requestHub.pushScope(); // TODO don't
```

Comment:
`pushScope` by itself here wouldn't be problematic, however combining this with e.g. setting `request` on the `requestHub` reference, leads to unexpected results where e.g. `request` isn't set on captured messages.

Reply:
Instead of `pushScope`, maybe we could simply call `requestHub.forkedCurrentScopes()` and then call `.makeCurrent()` on that. Then from there on only use the reference, returned by the fork call.

Reply:
I'm thinking tho we don't even need to call `pushScope` here since we're creating a new `Scopes` anyways.

Reply:
We could look into `Mono.using()` to do the cleanup for us in a follow up PR